packages/contracts-bedrock/test/L2/GasPriceOracle.t.sol

```diff
@@ -93,3 +102,111 @@ contract GasPriceOracle_Test is CommonTest {
+        bytes memory callDataPacked = abi.encodePacked(
+            basefeeScalar,
+            blobBasefeeScalar,
+            sequenceNumber,
+            timestamp,
+            number,
+            basefee,
+            blobBasefee,
+            hash,
+            batcherHash
+        );
+        bytes memory functionCallData = abi.encodePacked(functionSignature, callDataPacked);
+
+        // Execute the function call
+        vm.prank(depositor);
+        (bool success, ) = address(l1Block).call(functionCallData);
+        require(success, "Function call failed");
+
+        vm.prank(depositor);
+        gasPriceOracle.setEcotone();
+    }
+
+    /// @dev Tests that `setEcotone` is only callable by the depositor.
+    function test_setEcotone_wrongCaller_reverts() external {
+        vm.expectRevert("GasPriceOracle: only the depositor account can set isEcotone flag");
+        gasPriceOracle.setEcotone();
+    }
+
+    /// @dev Tests that `gasPrice` is set correctly.
+    function test_gasPrice_succeeds() external {
+        vm.fee(100);
+        uint256 gasPrice = gasPriceOracle.gasPrice();
+        assertEq(gasPrice, 100);
+    }
+
+    /// @dev Tests that `baseFee` is set correctly.
+    function test_baseFee_succeeds() external {
+        vm.fee(64);
+        uint256 gasPrice = gasPriceOracle.baseFee();
+        assertEq(gasPrice, 64);
+    }
+
+    /// @dev Tests that `overhead` reverts since it was removed in ecotone.
+    function test_overhead_legacyFunction_reverts() external {
+        vm.expectRevert("GasPriceOracle: overhead() is deprecated");
+        gasPriceOracle.overhead();
+    }
+
+    /// @dev Tests that `scalar` reverts since it was removed in ecotone.
+    function test_scalar_legacyFunction_reverts() external {
+        vm.expectRevert("GasPriceOracle: scalar() is deprecated");
+        gasPriceOracle.scalar();
+    }
+
+    /// @dev Tests that `l1BaseFee` is set correctly.
+    function test_l1BaseFee_succeeds() external {
+        assertEq(gasPriceOracle.l1BaseFee(), basefee);
+    }
+
+    /// @dev Tests that `blobBasefee` is set correctly.
+    function test_blobBasefee_succeeds() external {
+        assertEq(gasPriceOracle.blobBasefee(), blobBasefee);
+    }
+
+    /// @dev Tests that `basefeeScalar` is set correctly.
+    function test_basefeeScalar_succeeds() external {
+        assertEq(gasPriceOracle.basefeeScalar(), basefeeScalar);
+    }
+
+    /// @dev Tests that `blobBasefeeScalar` is set correctly.
+    function test_blobBasefeeScalar_succeeds() external {
+        assertEq(gasPriceOracle.blobBasefeeScalar(), blobBasefeeScalar);
+    }
+
+    /// @dev Tests that `decimals` is set correctly.
+    function test_decimals_succeeds() external {
+        assertEq(gasPriceOracle.decimals(), 6);
+        assertEq(gasPriceOracle.DECIMALS(), 6);
+    }
+
+    /// @dev Tests that `setGasPrice` reverts since it was removed in bedrock.
+    function test_setGasPrice_doesNotExist_reverts() external {
```

Comment:
This test can be omitted since its covered in the bedrock test

Reply:
removed